master/buildbot/worker/kubernetes.py

```diff
@@ -33,7 +33,10 @@ class KubeLatentWorker(DockerBaseWorker):
 
     instance = None
 
-    def default_pod_spec(self):
+    @defer.inlineCallbacks
+    def getPodSpec(self, build):
+        image = yield build.render(self.image)
```

Comment:
I guess this will susceptible to the same problem as hyper workers, which executed builds that weren't supposed to be executed on certain workers under high load conditions.

Reply:
indeed. thanks for reminder

Reply:
I pushed quick proposal for that: https://github.com/buildbot/buildbot/pull/4531/commits/19bfa95179b1ffcd25c53cf69a169a9bab828548

Reply:
Looks good.